test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,311 @@
+package images
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+// getPods looks up pods with "oc get" applying a label selector and returns
+// the names of matching pods
+func getPods(oc *exutil.CLI, selector string) ([]string, error) {
+	outPods, err := oc.Run("get").Args("pods",
+		"-l", selector,
+		"-o", "template",
+		"-t", "{{range .items}}{{.metadata.name}},{{end}}").Output()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.Trim(outPods, ","), ","), nil
+}
+
+// pollForPods will call getPods in a loop until the required number of pods
+// is returned
+func pollForPods(oc *exutil.CLI, selector string, count int, timeout time.Duration) ([]string, error) {
```

Comment:
should this function be placed in  `framework.go` or a standalone helper so it can be reused in other tests ?